tests/integration/test_charm.py

```diff
@@ -18,6 +18,7 @@ async def test_build_and_deploy(ops_test: OpsTest):
     charm = await ops_test.build_charm(".")
     await ops_test.model.deploy("postgresql-k8s")
     await ops_test.model.deploy(charm)
+    await ops_test.model.wait_for_idle(raise_on_error=False)
```

Comment:
Out of curiosity, what is the reason behind `raise_on_error=False`?

Reply:
The charm enters an error state sometimes/multiple times before stabilizing in a "waiting for database" status. I'll hope to remove this kind of behavior with the rewrite